solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder/src/test/java/com/solace/spring/cloud/stream/binder/SolaceBinderSubscriptionsOnNonDurableQueueIT.java

```diff
@@ -0,0 +1,129 @@
+package com.solace.spring.cloud.stream.binder;
+
+import com.solace.spring.boot.autoconfigure.SolaceJavaAutoConfiguration;
+import com.solace.spring.cloud.stream.binder.properties.SolaceConsumerProperties;
+import com.solace.spring.cloud.stream.binder.test.util.IgnoreInheritedTests;
+import com.solace.spring.cloud.stream.binder.test.util.InheritedTestsFilteredRunner;
+import com.solace.spring.cloud.stream.binder.test.util.SolaceTestBinder;
+import com.solacesystems.jcsmp.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.cloud.stream.binder.Binding;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.config.BindingProperties;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.util.MimeTypeUtils;
+
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests addition of topic subscriptions on non-durable queues.
+ * Non-Durable queues tests don't fit in SolaceBinderSubscriptionsIT since non-durable queues
+ * are not returned by SEMP. This class uses a different approach to validate correctness.
```

Comment:
SEMP returns non-durable queues. See [SolaceBinderBasicIT.testAnonConsumerDiscard()](https://github.com/SolaceProducts/solace-spring-cloud/blob/2.2.1/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder/src/test/java/com/solace/spring/cloud/stream/binder/SolaceBinderBasicIT.java#L324-L325)

Reply:
Thanks for that. I was wrongly using the sempV2Api.config() API instead of sempV2Api.monitor() API.
I simplified the tests accordingly.